### PR TITLE
Feat/create next app

### PR DESCRIPTION
**Pull request by NinaKammerlander**
### <strong>Pull Request</strong>
Added a next app as specified in #1075 and added a top nav for the next section.

<hr>

#### Type of PR
 Feature (non-breaking change which adds functionality)

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: apps/barista-design-system/src/app/app.module.ts [hunk 1]
```diff
@@ -30,8 +30,8 @@ import { DtAutocompleteModule } from '@dynatrace/barista-components/autocomplete
 import { DtTagModule } from '@dynatrace/barista-components/tag';
 import { DtThemingModule } from '@dynatrace/barista-components/theming';
 import { environment } from '../environments/environment';
-import { BaPageGuard } from '../shared/services/page-guard';
-import { BaPageService } from '../shared/services/page.service';
+import { BaPageGuard } from 'libs/shared/data-access-strapi/src/lib/page-guard';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/app/app.module.ts:34]: Use the tsconfig path alias for importing from a library.

```suggestion
import { BaPageGuard, BaPageService } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/app/app.server.module.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ import {
 import { AppModule } from './app.module';
 import { BaApp } from './app';
 import { HTTP_INTERCEPTORS } from '@angular/common/http';
-import { UniversalInterceptor } from '../shared/services/universal.interceptor';
+import { UniversalInterceptor } from 'libs/shared/data-access-strapi/src/lib/universal.interceptor';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/app/app.server.module.ts:25]:
```suggestion
import { UniversalInterceptor } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/app/app.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ import {
   BaPageService,
   getPageKeyFromUrl,
   getUrlPathName,
-} from '../shared/services/page.service';
+} from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/app/app.ts:38]:
```suggestion
} from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/app/components/nav/nav.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
 import { Component } from '@angular/core';
 import { BaNav as Navigation } from '@dynatrace/shared/barista-definitions';
-import { BaPageService } from '../../../shared/services/page.service';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/app/components/nav/nav.ts:19]:
```suggestion
import { BaPageService } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/decision-graph-page/components/ba-decision-graph/ba-decision-graph.spec.ts [hunk 5]
```diff
@@ -17,7 +17,7 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 
 import { BaDecisionGraph } from './ba-decision-graph';
-import { BaPageService } from '../../../../shared/services/page.service';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/decision-graph-page/components/ba-decision-graph/ba-decision-graph.spec.ts:20]:
```suggestion
import { BaPageService } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/decision-graph-page/components/ba-decision-graph/ba-decision-graph.module.ts [hunk 6]
```diff
@@ -23,7 +23,7 @@ import {
   BaDecisionGraphNode,
   BaDecisiongraphNodeNavigation,
 } from './ba-decision-graph-node';
-import { BaPageService } from '../../../../shared/services/page.service';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/decision-graph-page/components/ba-decision-graph/ba-decision-graph.module.ts:26]:
```suggestion
import { BaPageService } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/.search-page/search-page.module.ts [hunk 7]
```diff
@@ -17,7 +17,7 @@
 import { CommonModule } from '@angular/common';
 import { NgModule } from '@angular/core';
 import { Route, RouterModule } from '@angular/router';
-import { BaPageGuard } from '../../shared/services/page-guard';
+import { BaPageGuard } from '../../../../../libs/shared/data-access-strapi/src/lib/page-guard';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/.search-page/search-page.module.ts:20]:
```suggestion
import { BaPageGuard } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/decision-graph-page/components/ba-decision-graph/ba-decision-graph.ts [hunk 8]
```diff
@@ -16,8 +16,9 @@
 
 import { Component, OnInit } from '@angular/core';
 import { BaUxdNode } from '@dynatrace/shared/barista-definitions';
-import { BaPageService } from '../../../../shared/services/page.service';
-import { cloneDeep } from 'lodash-es';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
+// TODO: Use lodash-es instead. PR: https://github.com/dynatrace-oss/barista/pull/846
+import { cloneDeep } from 'lodash';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/decision-graph-page/components/ba-decision-graph/ba-decision-graph.ts:21]: why can we not use `lodas-es` here?

File: apps/barista-design-system/src/pages/decision-graph-page/components/ba-decision-graph/ba-decision-graph.ts [hunk 9]
```diff
@@ -16,8 +16,9 @@
 
 import { Component, OnInit } from '@angular/core';
 import { BaUxdNode } from '@dynatrace/shared/barista-definitions';
-import { BaPageService } from '../../../../shared/services/page.service';
-import { cloneDeep } from 'lodash-es';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/decision-graph-page/components/ba-decision-graph/ba-decision-graph.ts:19]:
```suggestion
import { BaPageService } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/decision-graph-page/decision-graph-page.module.ts [hunk 10]
```diff
@@ -20,7 +20,7 @@ import { Route, RouterModule } from '@angular/router';
 import { DtButtonModule } from '@dynatrace/barista-components/button';
 import { DtThemingModule } from '@dynatrace/barista-components/theming';
 import { BaDecisionGraphPage } from './decision-graph-page';
-import { BaPageGuard } from '../../shared/services/page-guard';
+import { BaPageGuard } from 'libs/shared/data-access-strapi/src/lib/page-guard';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/decision-graph-page/decision-graph-page.module.ts:23]:
```suggestion
import { BaPageGuard } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/decision-graph-page/decision-graph-page.ts [hunk 11]
```diff
@@ -24,7 +24,7 @@ import {
   ViewContainerRef,
   Injector,
 } from '@angular/core';
-import { BaPageService } from '../../shared/services/page.service';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/decision-graph-page/decision-graph-page.ts:27]:
```suggestion
import { BaPageService } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/error-page/error-page.ts [hunk 12]
```diff
@@ -15,7 +15,7 @@
  */
 
 import { Component } from '@angular/core';
-import { BaPageService } from '../../shared/services/page.service';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/error-page/error-page.ts:18]:
```suggestion
import { BaPageService } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/index-page/index-page.module.ts [hunk 13]
```diff
@@ -17,7 +17,7 @@
 import { CommonModule } from '@angular/common';
 import { NgModule } from '@angular/core';
 import { Route, RouterModule } from '@angular/router';
-import { BaPageGuard } from '../../shared/services/page-guard';
+import { BaPageGuard } from 'libs/shared/data-access-strapi/src/lib/page-guard';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/index-page/index-page.module.ts:20]:
```suggestion
import { BaPageGuard } from ''@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/index-page/index-page.ts [hunk 14]
```diff
@@ -15,7 +15,7 @@
  */
 
 import { Component } from '@angular/core';
-import { BaPageService } from '../../shared/services/page.service';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/index-page/index-page.ts:18]:
```suggestion
import { BaPageService } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/overview-page/overview-page.module.ts [hunk 15]
```diff
@@ -17,7 +17,7 @@
 import { CommonModule } from '@angular/common';
 import { NgModule } from '@angular/core';
 import { Route, RouterModule } from '@angular/router';
-import { BaPageGuard } from '../../shared/services/page-guard';
+import { BaPageGuard } from 'libs/shared/data-access-strapi/src/lib/page-guard';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/overview-page/overview-page.module.ts:20]:
```suggestion
import { BaPageGuard } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/shared/services/recently-ordered.service.ts [hunk 16]
```diff
@@ -19,7 +19,7 @@ import { DOCUMENT } from '@angular/common';
 import { Platform } from '@angular/cdk/platform';
 
 import { BaPageLink } from '@dynatrace/shared/barista-definitions';
-import { getUrlPathName } from './page.service';
+import { getUrlPathName } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/shared/services/recently-ordered.service.ts:22]:
```suggestion
import { getUrlPathName } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/single-page/single-page.ts [hunk 17]
```diff
@@ -24,7 +24,7 @@ import {
 import {
   BaPageService,
   getUrlPathName,
-} from '../../shared/services/page.service';
+} from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/single-page/single-page.ts:27]:
```suggestion
} from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/single-page/single-page.module.ts [hunk 18]
```diff
@@ -22,7 +22,7 @@ import { DtIconModule } from '@dynatrace/barista-components/icon';
 import { DtInputModule } from '@dynatrace/barista-components/input';
 import { DtTagModule } from '@dynatrace/barista-components/tag';
 import { BaComponentsModule } from '../../components';
-import { BaPageGuard } from '../../shared/services/page-guard';
+import { BaPageGuard } from 'libs/shared/data-access-strapi/src/lib/page-guard';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/single-page/single-page.module.ts:25]:
```suggestion
import { BaPageGuard } from '@dynatrace/shared/data-access-strapi';
```

File: apps/barista-design-system/src/pages/overview-page/overview-page.ts [hunk 19]
```diff
@@ -25,7 +25,7 @@ import {
 } from '@angular/core';
 import { _readKeyCode } from '@dynatrace/barista-components/core';
 import { fromEvent, Subscription } from 'rxjs';
-import { BaPageService } from '../../shared/services/page.service';
+import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/barista-design-system/src/pages/overview-page/overview-page.ts:28]:
```suggestion
import { BaPageService } from '@dynatrace/shared/data-access-strapi';
```

File: apps/next/browserslist [hunk 20]
```diff
@@ -0,0 +1,12 @@
...
+# https://github.com/browserslist/browserslist#queries
+
+# You can see what browsers were selected by your queries by running:
+#   npx browserslist
+
+> 0.5%
+last 2 versions
+Firefox ESR
+not dead
+not IE 9-11 # For IE 9-11 support, remove 'not'.
```
Comment (lukasholzer) [inline apps/next/browserslist:12]: I think we don't need to support IE in general only edge

File: apps/next/src/app/components/nav/nav.ts [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+  selector: 'next-nav',
+  templateUrl: 'nav.html',
+  styleUrls: ['nav.scss'],
+  host: {
+    class: 'next-nav',
+  },
+})
+export class Nav {
+  /** @internal Data needed to render the navigation. */
+  // _navData$ = this._pageService._getPage('nav');
```
Comment (lukasholzer) [inline apps/next/src/app/components/nav/nav.ts:32]: please remove unused code
```suggestion
```

File: apps/next/src/app/components/nav/nav.ts [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component } from '@angular/core';
+import { ThemeService } from '../../theme.service';
+// import { BaNav as Navigation } from '@dynatrace/shared/barista-definitions';
+// import { BaPageService } from 'libs/shared/data-access-strapi/src/lib/page.service';
```
Comment (lukasholzer) [inline apps/next/src/app/components/nav/nav.ts:20]:
```suggestion
```

File: apps/next/src/app/components/nav/nav.ts [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+
+@Component({
+  selector: 'next-nav',
+  templateUrl: 'nav.html',
+  styleUrls: ['nav.scss'],
+  host: {
+    class: 'next-nav',
+  },
+})
+export class Nav {
```
Comment (lukasholzer) [inline apps/next/src/app/components/nav/nav.ts:30]: please add Change detection strategy `OnPush`

File: apps/next/src/app/app.component.ts [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+
+import { Component } from '@angular/core';
+import { ThemeService } from './theme.service';
+
+@Component({
+  selector: 'x-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss'],
+})
+export class AppComponent {
```
Comment (lukasholzer) [inline apps/next/src/app/app.component.ts:25]: please add Change detection strategy `OnPush`

File: apps/next/src/app/components/sidenav/sidenav.ts [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+  /** Contains all section for a corresponding page group */
+  @Input()
+  set sections(sections: String[]) {
+    this._sections = sections;
+  }
+  _sections: String[];
+
+  constructor() {}
+
+  ngOnInit(): void {}
```
Comment (lukasholzer) [inline apps/next/src/app/components/sidenav/sidenav.ts:33]: please remove unused code

```suggestion
```

File: apps/next/src/app/pages/single-page/page-content.ts [hunk 26]
```diff
@@ -0,0 +1,93 @@
...
+    this._componentFactories = BA_CONTENT_COMPONENTS.map((componentType) =>
+      this._componentFactoryResolver.resolveComponentFactory(componentType),
+    );
+  }
+
+  ngOnDestroy(): void {
+    this._destroyContentComponents();
+  }
+
+  private _createContentComponents(): void {
```
Comment (lukasholzer) [inline apps/next/src/app/pages/single-page/page-content.ts:68]: can we extract this as this is a code duplication of the barista app? kind of a class that gets provided in the component that handles the kind of state of initializing the components.

Comment (ffriedl89) [reply apps/next/src/app/pages/single-page/page-content.ts:68]: This is already covered in the #1197. We moved a lot of code into a shared library.

File: apps/next/src/app/components/nav/nav.scss [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+
+.next-nav-link {
+  text-decoration: none;
+  padding-bottom: $fluid-spacing--2x-small;
+}
+
+.next-barista-theme {
+  display: flex;
+  align-items: center;
+  padding-right: 1rem;
```
Comment (ffriedl89) [inline apps/next/src/app/components/nav/nav.scss:53]: NIT: This should also use a spacing coming from the tokens.

File: apps/next/src/app/components/nav/nav.scss [hunk 28]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+.next-back-to-barista {
+  text-decoration: none;
+  text-transform: uppercase;
+  font-size: small;
```
Comment (ffriedl89) [inline apps/next/src/app/components/nav/nav.scss:59]: NIT: Not certain that font-size is already in the tokens - if so this should also use it.

File: apps/next/src/app/components/nav/nav.scss [hunk 29]
```diff
@@ -0,0 +1,64 @@
...
+  display: flex;
+  justify-content: flex-start;
+  padding-left: $fluid-spacing--small;
+  list-style-type: none;
+}
+
+.next-nav-brand {
+  font-weight: bold;
+  padding: $fluid-spacing--small;
+  font-size: large;
```
Comment (ffriedl89) [inline apps/next/src/app/components/nav/nav.scss:40]: NIT: Use tokens if possible

File: apps/next/src/app/shared/next-definitions.ts [hunk 30]
```diff
@@ -0,0 +1,20 @@
+/** Main navigation */
```
Comment (ffriedl89) [inline apps/next/src/app/shared/next-definitions.ts:1]: This will be removed in the future

File: apps/next/src/app/theme.service.ts [hunk 31]
```diff
@@ -0,0 +1,41 @@
...
+})
+export class ThemeService {
+  private _currentTheme$ = new BehaviorSubject<Theme>('abyss');
+
+  currentTheme$ = this._currentTheme$.asObservable();
+  currentThemeCssClass$ = this._currentTheme$.pipe(
+    map((theme) => `fluid-theme--${theme}`),
+  );
+
+  constructor() {}
```
Comment (ffriedl89) [inline apps/next/src/app/theme.service.ts:34]: NIT: Can be removed

File: apps/next/src/main.ts [hunk 32]
```diff
@@ -0,0 +1,31 @@
...
+import { AppModule } from './app/app.module';
+import { environment } from './environments/environment';
+
+if (environment.production) {
+  enableProdMode();
+}
+
+platformBrowserDynamic()
+  .bootstrapModule(AppModule, {
+    defaultEncapsulation: ViewEncapsulation.Emulated,
```
Comment (ffriedl89) [inline apps/next/src/main.ts:29]: NIT: Maybe this should be onPush

File: apps/next/src/styles.scss [hunk 33]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+button {
+  border: none;
+}
+
+button:focus {
+  outline: none;
+}
+
```
Comment (ffriedl89) [inline apps/next/src/styles.scss:64]: These will need some refactoring once we have proper components.

File: apps/next/src/tools/fetch-strapi-content.ts [hunk 34]
```diff
@@ -0,0 +1,89 @@
...
+
+interface BaristaContentListParams {
+  public?: boolean;
+  _limit?: number;
+}
+
+/**
+ * Fetches an array of elements from Strapi CMS.
+ */
+export async function fetchContentList<T extends NextPage>(
```
Comment (ffriedl89) [inline apps/next/src/tools/fetch-strapi-content.ts:32]: NIT: This should also be moved to the dataaccess library

File: libs/shared/data-access-strapi/tsconfig.lib.prod.json [hunk 35]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "./tsconfig.lib.json",
+  "angularCompilerOptions": {
+    "enableIvy": false
+  }
```
Comment (lukasholzer) [inline libs/shared/data-access-strapi/tsconfig.lib.prod.json:5]: please remove as the data access can use ivy

**Pull request merged**